src/TenancyServiceProvider.php

```diff
@@ -27,15 +27,16 @@ public function register(): void
         // Make sure Tenancy is stateful.
         $this->app->singleton(Tenancy::class);
 
-        // Make sure features are bootstrapped as soon as Tenancy is instantiated.
-        $this->app->extend(Tenancy::class, function (Tenancy $tenancy) {
+        $this->app->resolving(Tenancy::class, function (Tenancy $tenancy, $app) {
             foreach ($this->app['config']['tenancy.features'] ?? [] as $feature) {
-                $this->app[$feature]->bootstrap($tenancy);
+                $this->app[$feature]->bootstrap();
             }
-
-            return $tenancy;
         });
 
+        foreach ($this->app['config']['tenancy.tenant_unaware_features'] ?? [] as $feature) {
+            $this->app[$feature]->bootstrap();
+        }
```

Comment:
This is basically the main code of this PR, right?

We bootstrap tenant-aware features when `Tenancy` is actually used, and unaware features are bootstrapped **always**.

Could we get rid of the config and use a static method (maybe something like `alwaysBootstrap(): bool`) on the classes instead?

Also, what's the reason for the `extend` -> `resolving` change?

Reply:
> Could we get rid of the config and use a static method (maybe something like alwaysBootstrap(): bool) on the classes instead?

I tried this and it's working [fine](https://github.com/abrardev99/tenancy/commit/be6b2f82486d06db635008b823d1030137942ee0). I added this method in the interface so every class implement it.